swarm/pss/pss.go

```diff
@@ -365,17 +371,25 @@ func (self *Pss) process(pssmsg *PssMsg) bool {
 			self.outbox <- pssmsg
 		}()
 	}
-	handlers := self.getHandlers(psstopic)
+	if psstopic == rawTopic {
```

Comment:
I am not sure I understand this - how is it possible to end up in this if-condition? If we are using `rawTopic` for `psstopic` wouldn't we always end up in the `err != nil` block above?

This `if-condition` looks redundant to me.

Reply:
If the message is encrypted but has topic `0x0` it will hit this check.

Reply:
Then I don't understand the `raw` concept... from the description of this PR - `It also introduces a protected topic value 0x0 which, if set, denotes an unencrypted message, and is passed through to the raw message handler in case of failed decryption.`

Reply:
Sure we can, but the answer here is simple:

> passed through to the raw message handler in case of failed decryption

But if someone arbitrarily sets a topic `0x0` on an _encrypted_ message, we should catch that.

Ideally we would have zero-length topic. But since whisper topics are 4 bytes static arrays, this is an intermediate solution, at least until / if we detach from Whisper.

Reply:
You're right. I'll move the topic check up. We don't need to waste cycles trying to decrypt the message if it has topic 0x0